warehouse/query-core/src/main/java/datawave/query/planner/FederatedQueryPlanner.java

```diff
@@ -0,0 +1,315 @@
+                    throws DatawaveQueryException {
+        // Validate the config type.
+        if (!(genericConfig instanceof ShardQueryConfiguration)) {
+            throw new ClassCastException("Config must be an instance of " + ShardQueryConfiguration.class.getSimpleName());
+        }
+
+        ShardQueryConfiguration config = (ShardQueryConfiguration) genericConfig;
+
+        log.debug("Query originally set to execute against date range " + dateFormat.format(originalBeginDate) + "-" + dateFormat.format(originalEndDate));
+
+        // Get the relevant date ranges.
+        // TODO - Determine if we should pass in fields and datatypes to filter on for this query. Can we do this before calling process() on the query?
+        SortedSet<Pair<Date,Date>> dateRanges = getValidTargetDates(Collections.emptySet(), Collections.emptySet());
+
+        // TODO - Determine how restrictive we should be when evaluating whether or not to retain a date in the target date range, i.e. should we refrain from
+        // querying on a date if any index holes are seen on that day (current implementation) or only when we see index holes on a date for all fields and
+        // datatypes established for a query? What about when a query will include all fields and/or all datatypes?
+        if (dateRanges.isEmpty()) {
+            throw new DatawaveQueryException("No dates within query target date range exist without an index hole");
+        }
+
+        // Execute the same query for each date range and collect the results.
+        FederatedQueryIterable results = new FederatedQueryIterable();
+        int totalProcessed = 1;
+        for (Pair<Date,Date> dateRange : dateRanges) {
+            TraceStopwatch stopwatch = config.getTimers()
+                            .newStartedStopwatch("FederatedQueryPlanner - Execute query against date range subset " + dateFormat.format(dateRange.getLeft())
+                                            + "-" + dateFormat.format(dateRange.getRight()) + " [" + totalProcessed + " of " + dateRanges.size() + "]");
+            log.debug("Executing query against date range " + dateFormat.format(dateRange.getLeft()) + "-" + dateFormat.format(dateRange.getRight()));
+
+            // Set the new date range in a copy of the config.
+            ShardQueryConfiguration configCopy = new ShardQueryConfiguration(config);
+            configCopy.setBeginDate(dateRange.getLeft());
+            configCopy.setEndDate(dateRange.getRight());
+
+            // Create a copy of the original default query planner, and process the query with the new date range.
+            DefaultQueryPlanner planner = new DefaultQueryPlanner(originalPlanner);
+            results.addIterable(planner.process(config, query, settings, scannerFactory));
+
+            // Update the planned script.
+            // TODO - Verify if this is the way we want to consolidate the planned script for the multiple queries, and if ; works as a delimiter.
```

Comment:
This is a good question.  Luke has worked on having the subplans added to the query metrics.  Perhaps that is enough to be able to see all of the variants.  I think just using the first query plan as the "planned script" for the federated query logic is probably sufficient.

Reply:
Understood, I'll do that then.